rust/avdschema/src/utils/schema_from_path.rs

```diff
@@ -74,31 +75,25 @@ impl SchemaKeys {
 
         schema_keys.keys.extend(
             dict_schema
-                .dynamic_keys
-                .as_ref()
+                .get_dynamic_keys(dict)
                 .map(|dynamic_keys| {
                     dynamic_keys
-                        .keys()
-                        .flat_map(|dynamic_key_path| {
-                            get_dynamic_keys(dynamic_key_path, dict)
-                                .iter()
-                                .map(|dynamic_key| {
-                                    (
-                                        dynamic_key.to_owned(),
-                                        SchemaKey::DynamicKey {
-                                            dynamic_key_path: dynamic_key_path.to_owned(),
-                                        },
-                                    )
-                                })
-                                .collect::<Vec<_>>()
-                        })
-                        .collect::<Vec<_>>()
-                })
-                .unwrap_or_default(),
+                        .iter()
+                        .map(|(key, dynamic_key_info)| {
+                            (key.to_owned(), SchemaKey::from(dynamic_key_info))
+                        }).collect::<Vec<_>>()
+                }).into_iter().flatten()
+
```

Comment:
here we create an unecessary internediate Vec:


```suggestion
                .get_dynamic_keys(dict)
                .unwrap_or_default()
                .into_iter()
                .map(|key, dynamic_keys_info| {
                     (key.to_owned(), SchemaKey::from(&dynamic_key_info))
                })
```

Reply:
done and also improved the one above.